bazel/p4_library.bzl

```diff
@@ -120,3 +120,103 @@ p4_library = rule(
     incompatible_use_toolchain_transition = True,
     toolchains = ["@bazel_tools//tools/cpp:toolchain_type"],
 )
+
+def _p4_graphs_impl(ctx):
```

Comment:
It looks like this is duplicating a lot of code from _p4_library_impl. Could we reuse the code somehow?

Reply:
I added two helpers to reuse argument extraction and shell/cc environment setup